packages/next/src/root-layout/icons-metadata.ts

```diff
@@ -0,0 +1,140 @@
+import { QueryEngine } from '@snapwp/query';
+
+interface GeneralSettingsProps {
+	generalSettings: {
+		siteIcon: {
+			mediaItemUrl?: string;
+			mediaDetails: {
+				sizes?: IconData[];
+			};
+		};
+	};
+}
+
+interface IconMetaData {
+	faviconIcons: FormattedIconData[];
+	appleIcons: FormattedIconData[];
+	msApplicationTileIcon: IconData;
+}
+
+interface IconData {
+	sourceUrl: string;
+	height: string;
+	width: string;
+}
+
+interface FormattedIconData {
+	url: string;
+	sizes: string;
+}
+
+/**
+ * Fetch site icon data and filter them into categorized formats.
+ * @see - https://developer.wordpress.org/reference/functions/wp_site_icon/#:~:text=Displays%20site%20icon%20meta%20tags. Reason why we are different resolution icons into individual category.
```

Comment:
Documenting https://github.com/rtCamp/snapwp/pull/119#issuecomment-2736076355

```suggestion
 *
 * @see - https://developer.wordpress.org/reference/functions/wp_site_icon/#:~:text=Displays%20site%20icon%20meta%20tags. Reason why we are different resolution icons into individual category.
 *
 * @todo Refactor for composability alongside SEO metadata patterns 
```